### PR TITLE
feat: allow updating some fields on 12FA

### PR DESCRIPTION
**Pull request by madhuravius**
TODOs:

- [x] include terratest changes of updates

subsequent work (doing this outside the scope of this work) for additional testing (after more complex update backend pr has been merged in)

- [ ] add a tester role (similar to the acm-ar one) that has access to cloudwatch (read permissions for a bunch of resources)
- [ ] add a worker and then a test that checks cloudwatch logs
- [ ] add more test cases (including the multi-web example here: https://github.com/aptible/terraform-provider-aptible-v2/pull/28/files) with updates

---

Functions with updates:

image

image

---

Bulk of proof is in adjacent sweetness-v2 and aptible-api-v2 PRs

### CODE REVIEW

File: tests/twelve_factor_app/happy/twelve_factor_app_test.go [hunk 1]
```diff
@@ -12,43 +12,108 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func checkEnv(t *testing.T) {
+	for _, envVar := range []string{
+		"APTIBLE_AWS_ACCOUNT_ID",
+		"APTIBLE_USERNAME",
```
Comment (aguilinger) [inline tests/twelve_factor_app/happy/twelve_factor_app_test.go:18]: it's weird to set these explicityl when they're already set by the provider?

Comment (madhuravius) [reply tests/twelve_factor_app/happy/twelve_factor_app_test.go:18]: this is on the test here: https://github.com/aptible/terraform-provider-aptible-v2/pull/35/files#diff-04093d129c267cfd2e77d54517cd07f47008869fca86f101d7804211bb2a07f8R72

i preferred being more explicit than implicit here. i was running into issues with running `aptible logs` in one place and running against different environments' credentials in another.

File: tests/twelve_factor_app/examples/one_web_full/main.tf [hunk 2]
```diff
@@ -46,6 +41,14 @@ provider "aws" {
   }
 }
 
+provider "aptible" {
+  auth_host = var.auth_host
+  debug     = true
+  host      = var.api_host
+  username  = var.aptible_username
```
Comment (madhuravius) [inline tests/twelve_factor_app/examples/one_web_full/main.tf:48]: drop this here so it won't populate

File: tests/twelve_factor_app/examples/one_web_full/main.tf [hunk 3]
```diff
@@ -10,29 +10,24 @@ terraform {
...
-}
-
-variable "dns_host" {
-
-}
-variable "acm_aws_account" {
-
+variable "auth_host" {}
+variable "api_host" {}
+variable "aptible_username" { sensitive = true }
```
Comment (madhuravius) [inline tests/twelve_factor_app/examples/one_web_full/main.tf:15]: drop these two, we're gonna rely on env setting

File: tests/twelve_factor_app/happy/twelve_factor_app_test.go (hunk 1 above)
Comment (madhuravius) [reply tests/twelve_factor_app/happy/twelve_factor_app_test.go:18]: we're going to leave the env var guards but drop its assignment as terraform vars

File: tests/twelve_factor_app/happy/twelve_factor_app_test.go [hunk 4]
```diff
@@ -12,43 +12,108 @@ import (
...
+
 func TestTwelveFactorAppCreate(t *testing.T) {
+	checkEnv(t)
 	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
 		TerraformDir: "../examples/one_web_full",
 		Lock:         false,
 		Vars: map[string]interface{}{
+			"acm_aws_account":        os.Getenv("ACM_AWS_ACCOUNT_ID"),
 			"aptible_aws_account_id": os.Getenv("APTIBLE_AWS_ACCOUNT_ID"),
+			"aptible_username":       os.Getenv("APTIBLE_USERNAME"),
```
Comment (madhuravius) [inline tests/twelve_factor_app/happy/twelve_factor_app_test.go:72]: drop these `aptible_username`, `aptible_password`

**Comment by madhuravius** [comment]
Proof of function from this terratest change

initial create:

Screenshot 2023-02-17 at 2 54 48 PM

update:

Screenshot 2023-02-17 at 2 54 42 PM

Screenshot 2023-02-17 at 3 02 37 PM

teardown:

**Review by aguilinger — APPROVED** [review]

**Pull request merged**